firefox_ui_tests/remote/security/test_untrusted_connection_error_page.py

```diff
@@ -0,0 +1,35 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import By
+from marionette_driver.errors import MarionetteException
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestUntrustedConnectionErrorPage(FirefoxTestCase):
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.test_url = 'https://ssl-selfsigned.mozqa.com'
+
+    def tearDown(self):
+        FirefoxTestCase.tearDown(self)
+
+    def test_untrusted_connection_error_page(self):
+        # Test the GetMeOutOfHereButton from an Untrusted Error page
+        with self.marionette.using_context('content'):
+            self.assertRaises(MarionetteException,
+                              self.marionette.navigate, self.test_url)
+            # Wait for the DOM to receive events
+            time.sleep(1)
+
+            button = self.marionette.find_element(
+                By.ID, "getMeOutOfHereButton")
+            self.assertTrue(button.is_displayed())
+            button.click()
+            self.wait_for_condition(
+                lambda mn: self.browser.default_homepage in mn.get_url())
```

Comment:
This will not work for locales which have a redirect as their default home page. What we have to do here is to wait until the target page has been loaded, whereby this is still a missing piece in our tests. So maybe better is to run `self.browser.get_final_url(self.browser.default_homepage)` at the very first in this test and use the target page as reference for this check, which can then even be a `==` and not `in`.